src/hypxiel_api_python/utils/requests.py

```diff
@@ -1,34 +1,63 @@
-from aiohttp import ClientSession
 from datetime import datetime
+from aiohttp import ClientSession
+from src.hypxiel_api_python.utils.cache import purge_cache
+from src.hypxiel_api_python.hypixel import HypixelAPI
 
 
-async def get(self, endpoint: str, param: tuple, nocache: bool = None):
-    api = self.hypixel_api
+async def get(hypixel_api: HypixelAPI, endpoint: str, params: dict[str, str], nocache: bool = None) -> dict:
     """Determine whether it should use a cached object or fetch a new one."""
     """`nocache` is a way for the user to bypass the cache"""
-    cached_value = self.hypixel_api.endpoints[param][endpoint.replace("/", "")]
-    if not cached_value:
-        return await fetch(self=self, url=f"{api.path}/{endpoint}?key={api.key}&{param[0]}={param[1]}")
+    if not endpoint.startswith("/"):
+        endpoint = f"/{endpoint}"
```

Comment:
is this really necessary?

Reply:
Was also thinking about using an enum for endpoint but idk. is that a good idea?